src/main/java/algorithms/curated170/medium/NumberOfDistinctIslands.java

```diff
@@ -0,0 +1,88 @@
+package algorithms.curated170.medium;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class NumberOfDistinctIslands {
+    
+    Set<String> islands;
+    int[][] grid;
+    int xDim;
+    int yDim;
+    public int numOfIslands(int[][] grid)
+    {
+        this.grid = grid;
+        islands = new HashSet<>();
+        xDim = grid.length;
+        yDim = grid[0].length;
+
+        for(int i = 0; i<xDim; i++)
+        {
+            for(int j = 0; j<yDim; j++)
+            {
+                String k = DFS(i, j);
+                if(k!="#")
+                {
+                    islands.add(k);
+                }
+            }
+        }
+
+        return islands.size();
+    }
+    
+    private String DFS(int i, int j)
+    {
+        String islandCode = "";
+        if(grid[i][j] == 0)
+        {
+            return "#";
+        }
+        grid[i][j] = 0;
+        if(i<xDim-1&&grid[i+1][j] == 1)
```

Comment:
We should reduce if statements into a single base case as we have done in:
 https://github.com/spiralgo/algorithms/blob/master/src/main/java/algorithms/medium/NumberOfIslands.java